### PR TITLE
Looking for initial feedback from Dan

### PR DESCRIPTION
**Pull request by AlexanderLiteplo**
@DanL0 
Looking for feedback in general but would be great to get your opinion on how to test the edge case where
```
                if (_input._nativeDrop != 0) {
                    require(_input._nativeDrop <= type(uint128).max, "OFTWrapper: nativeDrop exceeds uint128 max");
                    options = OptionsBuilder.newOptions().addExecutorNativeDropOption(
                        uint128(_input._nativeDrop),
                        _input._toAddress
                    );
                }
```
Currently I have a very hacky way of testing this using a Mock. I tried the whole day to find a better way and the only other way I could think of was decoding the options array. However that is a lot of work (unless there is a method that easily does this) and the current solution I have is good enough to test that options are created when native drop is > 0.

I figured I would ask your opinion before going deeper down the road of trying to decode options and then retrieve the nativeDrop from it and check that it is the correct value.

### CODE REVIEW

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 1]
```diff
@@ -475,91 +491,146 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
...
+    function _getOftLimitsAndReceiptsForEpv2(
+        QuoteInput calldata _input,
+        FeeObj calldata _feeObj,
+        QuoteResult memory quoteResult,
+        uint256 amountAfterWrapperFees,
+        uint256 fees
+    ) internal view returns (OFTQuoteResult memory) {
+        (
+            OFTLimit memory oftLimit, // {minAmountLD, maxAmountLD}
+            // Ignore oftFeeDetails because those aren't implemented yet
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:587]: Is this ignored temporarily? Seems like OFT Fee should be part of what we return but please double check with Stargate team

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 2]
```diff
@@ -475,91 +491,146 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
...
+        (quoteResult.fees, amountAfterWrapperFees) = _calculateInitialFeesAndAmount(_input, _feeObj, quoteResult.fees);
+        fees = quoteResult.fees[0].amount + quoteResult.fees[1].amount;
 
         if (_input.version == 1) {
-            uint256 oftFee = Fee(_input._token).quoteOFTFee(_input._dstChainId, amountAfterWrapperFees);
-            bytes memory data = abi.encodeWithSignature(
-                "quoteOFTFee(uint16,uint256)",
-                _input._dstChainId,
-                amountAfterWrapperFees
+            // uint256 oftFee = Fee(_input._token).quoteOFTFee(_input._dstEid, amountAfterWrapperFees);
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:500]: Is this ignored temporarily? Seems like OFT Fee should be part of what we return but please double check with Stargate team

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 3]
```diff
@@ -475,91 +491,146 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
...
+            OFTQuoteResult memory oftQuoteResult = _getOftLimitsAndReceiptsForEpv2(
+                _input,
+                _feeObj,
+                quoteResult,
+                amountAfterWrapperFees,
+                fees
+            );
+            quoteResult = oftQuoteResult.quoteResult;
+            fees = oftQuoteResult.fees;
+            amountAfterWrapperFees = oftQuoteResult.amountAfterWrapperFees;
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:536]: What is the point of assigning value to this variable here if it isn't used later?

File: packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol [hunk 4]
```diff
@@ -777,32 +784,171 @@ contract OFTWrapperTest is Test, LzTestHelper {
...
+        assertEq(quoteResult.srcAmountMin, MOCK_MIN_AMOUNT_LD);
+        assertEq(quoteResult.amountReceivedLD, mockAmountReceivedLD);
+
+        uint256 expectedSrcAmount = mockAmountSentLD + expectedWrapperFee + expectedCallerFee;
+        assertEq(quoteResult.srcAmount, expectedSrcAmount);
+
+        logQuoteResults(quoteResult);
+    }
+
+    function logQuoteResults(IOFTWrapper.QuoteResult memory quoteResult) internal view {
```
Comment (DanL0) [inline packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol:945]: This might be useful for testing right now but in final PR make sure to remove these console logs for cleanliness of running test suite. Theoretically it could be behind an environment flag or some verbosity control.

File: packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol [hunk 5]
```diff
@@ -777,32 +784,171 @@ contract OFTWrapperTest is Test, LzTestHelper {
...
+
+        // Calculate expected fees
+        uint256 expectedWrapperFee = (input._amount * DEFAULT_BPS) / BPS_DENOMINATOR;
+        uint256 expectedCallerFee = (input._amount * feeObj.callerBps) / BPS_DENOMINATOR;
+        uint256 amountAfterWrapperFees = input._amount - expectedWrapperFee - expectedCallerFee;
+
+        // Set up mock return values
+        uint256 mockDust = 100;
+        uint256 mockAmountSentLD = amountAfterWrapperFees - mockDust;
+        // todo: check if this is correct given amountAfterWrapperFees is what is passed into debitview so
```
Comment (DanL0) [inline packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol:847]: For OFT Endpoint V2 I think Stargate team was happy with results from the initial code I added to OFTWrapper, so maybe for correctness of OFT Endpoint V2 values you could consult base branch

unless you have received feedback from Stargate team that it wasn't correct and it contained errors

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol [hunk 6]
```diff
@@ -22,20 +22,32 @@ interface IOFTWrapper {
...
         uint256 _nativeDrop;
     }
 
+    struct QuoteFee {
+        string fee;
+        uint256 amount;
+        address token;
+    }
+
+    struct OFTQuoteResult {
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol:38]: I don't see a good justification or separation of this interface from QuoteResult interface. Maybe you could instead change return values of `_getOftLimitsAndReceiptsForEpv2` to `(OFTQuoteResult memory. uint256 amountAfterWrapperFees, uint256 fees)` but use more descriptive name for `fees`?

Or maybe remove that additional fees return value and instead use QuoteResult interface (that you already use)?

The problem is you already have `fees` entry in `QuoteResult` interface but you don't seem to use it in `_getOftLimitsAndReceiptsForEpv2` but instead you duplicate it.

**Comment by DanL0** [comment]
For getting OFT V1 block confirmations probably you can look at documentation for setting them on EP V1 and figure out how to read them: https://docs.layerzero.network/v1/developers/evm/evm-guides/ua-custom-configuration/overview

File: packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/mocks/epv2/MockOFT.sol [hunk 7]
```diff
@@ -59,3 +59,43 @@ contract CustomQuoteSendMockOFTAdapter is MockOFTAdapter {
...
+        string memory _symbol,
+        address _lzEndpoint,
+        address _delegate
+    ) MockOFT(_name, _symbol, _lzEndpoint, _delegate) {}
+
+    OFTReceipt private oftReceipt;
+    OFTLimit private oftLimit;
+    MessagingFeeEpv2 private quoteSendFee;
+
+    uint256 public constant NATIVE_FEE_PRESENT = 1 ether; // Indicates native drop present
```
Comment (DanL0) [inline packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/mocks/epv2/MockOFT.sol:75]: if even using something like for testing I would consider changing it from `1` which seems like common input to something much more unlikely like `777777` or something.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 8]
```diff
@@ -475,91 +491,146 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
...
+                    IOAppCore(oftAddress).endpoint().getSendLibrary(oftAddress, _input._dstEid)
+                ).getConfig(_input._dstEid, oftAddress, 2);
                 UlnConfig memory ulnConfig = abi.decode(rawConfig, (UlnConfig));
                 quoteResult.confirmations = ulnConfig.confirmations;
             }
         }
     }
+
+    function _getOftLimitsAndReceiptsForEpv2(
+        QuoteInput calldata _input,
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:579]: Please change `QuoteInput` to not have _ in front of property names. I think I added it for no good reason.

**Comment by DanL0** [comment]
> Currently I have a very hacky way of testing this using a Mock. I tried the whole day to find a better way and the only other way I could think of was decoding the options array. However that is a lot of work (unless there is a method that easily does this) and the current solution I have is good enough to test that options are created when native drop is > 0.

vm.expectCall seems to work:
```diff
diff --git a/packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol b/packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol
index 3ca3f45e..f55e1e17 100644
--- a/packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol
+++ b/packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol
@@ -916,6 +916,28 @@ contract OFTWrapperTest is Test, LzTestHelper {
         );
         oft.setQuoteSendReturnValue(MessagingFeeEpv2({ nativeFee: MOCK_NATIVE_FEE, lzTokenFee: 0 }));
 
+        SendParamEpv2 memory b = SendParamEpv2({
+            dstEid: input._dstEid,
+            to: input._toAddress,
+            amountLD: 985299999999999999970, // 9.852e20
+            minAmountLD: 900000000000000000000, // 9e20
+            extraOptions: OptionsBuilder.newOptions().addExecutorNativeDropOption(
+                uint128(input._nativeDrop),
+                input._toAddress
+            ),
+            composeMsg: hex"",
+            oftCmd: hex""
+        });
+
+        // Expect the call to oft.quoteSend with the specified parameters
+        vm.expectCall(
+            address(oft),
+            abi.encodeWithSelector(
+                oft.quoteSend.selector,
+                b,
+                false // As per the stack trace
+            )
+        );
         IOFTWrapper.QuoteResult memory quoteResult = oftWrapper.quote(input, feeObj);
 
         assertEq(quoteResult.fees.length, 3);
```

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 9]
```diff
@@ -461,10 +464,23 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
         require(_callerBps <= callerBpsCap, "OFTWrapper: callerBps > callerBpsCap");
     }
 
-    // @TODO parametrize per token shared decimals and ld2sd2 rate for e v1
-    function _removeDust(uint _amount) internal view virtual returns (uint amountAfter, uint dust) {
-        dust = 0;
-        amountAfter = _amount - dust;
+    function exposed_removeDust(
```
[the remark this answers is not in the document]
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:467]: Yeah sorry I just have that in as a temporary thing for now

I'll be replacing it with the mock method for sure

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 10]
```diff
@@ -475,91 +491,146 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
 
         uint256 fees = 0;
         uint256 amountAfterWrapperFees = 0;
+        quoteResult.fees = new QuoteFee[](3);
```
[the remark this answers is not in the document]
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:494]: I don't think it would be considered an override. We are just allocating memory in the main function, then in the helper setting the allocated memory to the calculated fees. I think it would maybe not be good to create the QuoteFee array in the helper because the 3rd element gets set outside of the helper and to me that is confusing readability.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 1 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:587]: Actually this was something I'm confused about but you're right it would probably make sense to return.

Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:587]:
```    /**
     * @notice Provides a quote for OFT-related operations.
     * @param _sendParam The parameters for the send operation.
     * @return oftLimit The OFT limit information.
     * @return oftFeeDetails The details of OFT fees.
     * @return oftReceipt The OFT receipt information.
     */
    function quoteOFT(
        SendParam calldata _sendParam
    )
        external
        view
        virtual
        returns (OFTLimit memory oftLimit, OFTFeeDetail[] memory oftFeeDetails, OFTReceipt memory oftReceipt)
    {
        uint256 minAmountLD = 0; // Unused in the default implementation.
        uint256 maxAmountLD = type(uint64).max; // Unused in the default implementation.
        oftLimit = OFTLimit(minAmountLD, maxAmountLD);

        // Unused in the default implementation; reserved for future complex fee details.
        oftFeeDetails = new OFTFeeDetail[](0);

        // @dev This is the same as the send() operation, but without the actual send.
        // - amountSentLD is the amount in local decimals that would be sent from the sender.
        // - amountReceivedLD is the amount in local decimals that will be credited to the recipient on the remote OFT instance.
        // @dev The amountSentLD MIGHT not equal the amount the user actually receives. HOWEVER, the default does.
        (uint256 amountSentLD, uint256 amountReceivedLD) = _debitView(
            _sendParam.amountLD,
            _sendParam.minAmountLD,
            _sendParam.dstEid
        );
        oftReceipt = OFTReceipt(amountSentLD, amountReceivedLD);
    }```

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 2 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:500]: This is something we should chat about yes

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 1 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:587]: @AlexanderLiteplo get the fee details and return them. The logic is implemented.

File: packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol (hunk 5 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/test/unitest/peripheral/oft-wrapper/OFTWrapper.t.sol:850]: @DanL0 So are you saying we don't have to test correctness?

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 11]
```diff
@@ -473,93 +482,206 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
...
+            bytes memory options = bytes("");
+            if (_input.nativeDrop != 0) {
+                require(_input.nativeDrop <= type(uint128).max, "OFTWrapper: nativeDrop exceeds uint128 max");
+                options = OptionsBuilder.newOptions().addExecutorNativeDropOption(
+                    uint128(_input.nativeDrop),
+                    _input.toAddress
+                );
+            } else {
+                // TODO: make this a constant or a parameter
+                options = OptionsBuilder.newOptions().addExecutorLzReceiveOption(200_000, 0);
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:574]: you should never set these options in terms of just lzReceive gas

it should be set as EnforcedOptions before trying to quote an OFT

If someone wants nativedrop add native drop option like it is already in code
in the other case don't add any options and leave them empty, default enforced options should be used, we shouldn't add arbitrary amount of lzReceive gas for users in this task

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol [hunk 12]
```diff
@@ -19,23 +26,44 @@ interface IOFTWrapper {
...
-        uint256 _minAmount;
-        bytes32 _toAddress;
-        uint256 _nativeDrop;
+        OFTVersion version;
+        address token;
+        uint16 dstEid;
+        uint256 amountLD;
+        uint256 minAmountLD;
+        bytes32 toAddress;
+        bytes extraOptions;
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol:35]: please remove extraOptions and make it less generic by using previous interface here:

use nativeDrop instead of extraOptions:
```
uint256 _nativeDrop;
```

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 13]
```diff
@@ -473,93 +489,276 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
...
+            });
         }
 
-        if (_input.version == 1) {
-            uint256 oftFee = Fee(_input._token).quoteOFTFee(_input._dstChainId, amountAfterWrapperFees);
-            bytes memory data = abi.encodeWithSignature(
-                "quoteOFTFee(uint16,uint256)",
-                _input._dstChainId,
-                amountAfterWrapperFees
+        if (_input.amountLD > oftLimit.maxAmountLD) {
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:689]: It seems this code isn't doing anything meaningful according to AI and my initial look. AI response:
image

maybe you need to use oftLimit.maxAmountLD in place of srcAmount instead?

also, please do this check and transformation only if additional flag in QuoteInput is passed, because of the default quoteOFT implementation being only informative

new boolean flag might be called eg. QuoteInput.ignoreLimit and in most cases will be true

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol [hunk 14]
```diff
@@ -19,23 +26,44 @@ interface IOFTWrapper {
...
     struct QuoteResult {
         uint256 srcAmount;
-        uint256 dstAmount; // quoteOFT
+        uint256 amountReceivedLD;
         uint256 srcAmountMin;
         uint256 srcAmountMax;
-        uint256 nativeFee;
-        uint256 confirmations;
+        uint64 confirmations;
+        QuoteFee[] fees;
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol:66]: please extract `uint256 nativeFee;` from fees array so the native message fee is separate from token fees

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol [hunk 15]
```diff
@@ -473,93 +489,276 @@ contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
...
+        (quoteResult.srcAmountMax, ) = _removeDust(type(uint256).max, decimals, sharedDecimals);
+
+        quoteResult.srcAmount = quoteResult.amountReceivedLD + _input.wrapperAndCallersFees;
+
+        (uint256 nativeFee, ) = IOFTV2(_input.token).estimateSendFee(
+            _input.dstEid,
+            _input.toAddress,
+            quoteResult.amountReceivedLD,
+            false,
+            bytes("")
```
Comment (DanL0) [inline packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:576]: For all OFT on EpV1:

please add automatic fetching of minDstGasLimit and setting it in this place: https://docs.layerzero.network/v1/developers/evm/evm-guides/advanced/relayer-adapter-parameters

if nativeDrop is set also make sure that it is taken into account

**Review by DanL0 — CHANGES_REQUESTED** [review]
After addressing this feedback we can test the adapter on one of the chains.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol (hunk 12 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol:35]: Having extraOptions be passed in is a much cleaner solution than just native drop. It also follows the convention of all of the other methods in OFTWrapper. 

The main reason I've done this though is that, for the all of the integration tests (e.g.: `testQuote_OFTEpV2_Integration_With_SendEpv2`), if I don't add `bytes memory options = OptionsBuilder.newOptions().addExecutorLzReceiveOption(200_000, 0);` as options for the quoteInput function, I get the following error `[FAIL: InvalidWorkerOptions(0)] testQuote_OFTEpV2_Integration_With_SendEpv2() (gas: 6482002)`

I'm not entirely sure why this happens but we can hop on a call and I can demonstrate this behaviour to you.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 13 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:689]: Hey Dan, I think this is necessary.

Maybe we can triple check with Piotr but if I remember correctly he told me this check + recalculation was necessary.

Also here was the original instruction about this however it has a typo (as we discussed previously) where it should be amount > maxAmount rather than the following:
image

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol (hunk 14 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol:66]: fixed

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol (hunk 12 above)
Comment (DanL0) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol:35]: Did you set enforced options in your tests? This is usual reason for: `InvalidWorkerOptions(0)`

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 13 above)
Comment (DanL0) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:689]: Maybe you can create a test case that showcases how this logic is actually used? As it currently stands I think (+ AI too) that this code block is doing nothing from logical standpoint so would be great if you could demonstrate otherwise.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 15 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:576]: Would you also just be able to explain this comment in greater detail? I'm a bit confused as to what you want me to do. I understand I should add minDstGas for Epv1 OFTs but I'm not sure what you mean by "setting it". Also I don't think nativeDrop interacts with minDstGasLimit at all.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 13 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:689]: I messaged Piotr and I'll wait on his review for this. He was the one who suggested this check in the first place so I'll make sure he is ok with this being removed.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol (hunk 12 above)
Comment (AlexanderLiteplo) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/interfaces/IOFTWrapper.sol:35]: enforced worker options set fixes it thanks!

native drop is now directly passed.

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 15 above)
Comment (DanL0) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:576]: Just make sure to pass minDstGas required to underlying OFT thats it. I didn't mean setting minDstGas in the OApp. I meant using existing setting in the OFT Wrapper.

nativeDrop doesn't interact with minDstGasLimit but you pass nativeDrop as Relayer Adapter Params (https://docs.layerzero.network/v1/developers/evm/evm-guides/advanced/relayer-adapter-parameters#airdrop)

File: packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol (hunk 13 above)
Comment (DanL0) [reply packages/stg-evm-v2/src/peripheral/oft-wrapper/OFTWrapper.sol:689]: I looked at it again and I still think this code isn't doing anything. get amount and fees is called again but with exactly the same input which hasn't been changed so it will return exactly the same result. Whole _input object is the same as in initial call so it has to produce the same result.

I think having a test case for this would be beneficial otherwise its hard to justify this code fragment being there.
